### PR TITLE
Bug fix: #30 NewRealisClient method returning <nil> error when zookee…

### PR DESCRIPTION
**Pull request by rdelval**
…per leader not found

### CODE REVIEW

File: zk.go [hunk 1]
```diff
@@ -41,32 +41,32 @@ type NoopLogger struct{}
...
-		if err != nil {
-			fmt.Println("error in LeaderFromZK: ", err)
+
+		// Backoff if we failed to determine leader
+		adjusted := duration
+		if defaultBackoff.Jitter > 0.0 {
+			adjusted = Jitter(duration, defaultBackoff.Jitter)
 		}
+		fmt.Printf("Error determining Aurora leader: %v; retrying in %v\n", err, adjusted)
+		time.Sleep(adjusted)
```
Comment (misho-kr) [inline zk.go]: Why do you need to sleep after the last retry? It seems a waste of time.

Comment (rdelval) [inline zk.go]: In this case we're doing that so we don't accidentally DOS any service running. If we have multiple clients firing off jobs and querying ZK too quickly, if an error happens, we don't want them all to retry at the same time.

Comment (misho-kr) [inline zk.go]: Hmm, I am not sure you understood my concern. There is the _Jitter_ function that should prevent scores of clients hitting ZK at the same time by spreading the retries.

And there is the retry count which presumably is greater than 0, so DOS scenario should not develop.

You could increase the retry count by 1 to achieve the same effect (holding off clients from retrying too rapidly or too often) if that is your goal. OTOH with this extraneous sleep clients are forced to pause even when there is no chance a successful connection can be made.

Comment (rdelval) [inline zk.go]: OK, now I see what you're saying. Will submit a patch so that this exits if the last attempt is an error.

**Pull request merged**